File: csharp/Person.cs
namespace StudentManagementSystem
{
    // Abstract base class (equivalent of Python's ABC + abstractmethod)
    public abstract class Person
    {
        private int id;
        private string name;

        protected Person(int id, string name)
        {
            this.id = id;
            this.name = name;
        }

        // Encapsulation: getter/setter for name
        public string GetName()
        {
            return name;
        }

        public void SetName(string name)
        {
            this.name = name;
        }

        // Abstract method -> must be overridden by subclasses (polymorphism)
        public abstract void DisplayRole();
    }
}

File: python/person.py
from abc import ABC, abstractmethod

class Person(ABC):

    def __init__(self, id, name):
        self.__id = id
        self.__name = name

    # Getter
    def get_name(self):
        return self.__name

    # Setter
    def set_name(self, name):
        self.__name = name

    @abstractmethod
    def display_role(self):
        pass